### PR TITLE
examples: add gzip decompression

### PR DESCRIPTION
**Pull request by giantcroc**
Signed-off-by: giantcroc <changran.wang@intel.com>

Commit Message: add gzip decompression example
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/start/sandboxes/gzip.rst [hunk 1]
```diff
@@ -63,7 +64,29 @@ As only files with a content-type of ``application/json`` are configured to be g
...
 
-Step 3: Test compression of Envoy’s statistics
+Step 3: Test Envoy’s decompression of downstream files
+**************************************************
+
+The sandbox is configured with one endpoint on port ``10000`` for uploading downstream files:
+
+- ``/upload``
+
+Use ``wget`` to get compressed file ``file.gz``
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:74]: dont use wget please - we want to keep the deps/complexity to a minimum and curl is already used

also, im wondering if this example would work better sending the server a message with a fresh file

also, im not sure how this works - i think the file you are using - ie `file.gz` is not actually compressed

**Review by phlax — COMMENTED** [review]
i will review more thoroughly once tests are passing - but leaving an initial comment

File: docs/root/start/sandboxes/gzip.rst (hunk 1 above)
Comment (giantcroc) [reply docs/root/start/sandboxes/gzip.rst]: > dont use wget please - we want to keep the deps/complexity to a minimum and curl is already used
> 
> also, im wondering if this example would work better sending the server a message with a fresh file
> 
> also, im not sure how this works - i think the file you are using - ie `file.gz` is not actually compressed

`file.gz` is the data compressed by compressor, and I have tested it's size, I think it's good to implement a round-trip test.

Comment (phlax) [reply docs/root/start/sandboxes/gzip.rst]: cool - i confirmed that this works as described, and agree that works as a good example

**Review by phlax — APPROVED** [review]
great, thanks for this @giantcroc - really good improvement to this example

**Pull request merged**